modules/EvseV2G/charger/ISO15118_chargerImpl.cpp

```diff
@@ -354,19 +354,16 @@ void ISO15118_chargerImpl::handle_cableCheck_Finished(bool& status) {
 void ISO15118_chargerImpl::handle_set_Certificate_Service_Supported(bool& status) {
     if (status == true) {
         // For setting "Certificate" in ServiceList in ServiceDiscoveryRes
+        struct iso1ServiceType cert_service;
         const std::string service_name = "Certificate";
-        uint8_t len = service_name.length();
-        v2g_ctx->evse_v2g_data.evse_service_list[v2g_ctx->evse_v2g_data.evse_service_list_len].FreeService = 1;
-        v2g_ctx->evse_v2g_data.evse_service_list[v2g_ctx->evse_v2g_data.evse_service_list_len].ServiceID = 2;
-        v2g_ctx->evse_v2g_data.evse_service_list[v2g_ctx->evse_v2g_data.evse_service_list_len].ServiceCategory = iso1serviceCategoryType_ContractCertificate;
-        memcpy(v2g_ctx->evse_v2g_data.evse_service_list[v2g_ctx->evse_v2g_data.evse_service_list_len].ServiceName.characters, reinterpret_cast<const char*>(service_name.data()), len);
-        v2g_ctx->evse_v2g_data.evse_service_list[v2g_ctx->evse_v2g_data.evse_service_list_len].ServiceName.charactersLen = len;
-        v2g_ctx->evse_v2g_data.evse_service_list_len += 1;
-
-        // For setting "Installation" in the ServiceParameterList in ServiceDetailRes
-        configure_parameter_set(&v2g_ctx->evse_v2g_data.service_parameter_list[0], 1);
-        // TODO: support "Update" in the ServiceParameterList in ServiceDetailRes
-        // configure_parameter_set(&v2g_ctx->evse_v2g_data.service_parameter_list[0], 2);
+        const int16_t cert_parameter_set_id = 1; // parameter-set-ID 1: "Installation" service. TODO: Support of the "Update" service (parameter-set-ID 2)
```

Comment:
so the idea in the future will be creating an array of const int16_t to pass it to add_service_to_service_list? Why not finish the implementation now?

Reply:
because we are currently only supporting one parameter-set-id. Why we should create an array for one element?